tests/views/test_message.py

```diff
@@ -307,3 +311,32 @@ def test_conversation_subject_error(self, mock_request):
         response = self.app.get("/messages/threads/fb0e79bd-e132-4f4f-a7fd-5e8c6b41b9af")
         self.assertEqual(response.status_code, 200)
         self.assertIn("No Subject".encode(), response.data)
+
+    def test_get_radio_buttons(self):
+        response = self.app.get("/messages")
+
+        self.assertEqual(200, response.status_code)
+        self.assertIn("Filter messages by survey".encode(), response.data)
+        self.assertIn("ASHE".encode(), response.data)
+        self.assertIn("Bricks".encode(), response.data)
+
+    @requests_mock.mock()
+    def test_radio_buttons_post_nothing_selected(self, mock_request):
+        mock_request.get(url_get_threads_list, json=thread_list)
+        mock_request.get(url_get_surveys_list, json=self.surveys_list_json)
+
+        response = self.app.post("/messages", follow_redirects=True)
+
+        self.assertEqual(200, response.status_code)
+        self.assertIn("Home".encode(), response.data)
+        self.assertIn("Please select a survey to filter your messages.".encode(), response.data)
+        self.assertIn("Filter messages by survey".encode(), response.data)
+
+    @requests_mock.mock()
+    def test_get_messages_survey_does_not_exist(self, mock_request):
+        mock_request.get(shortname_url)
+
+        response = self.app.get("/messages/this-survey", follow_redirects=True)
+
+        self.assertEqual(response.status_code, 500)
```

Comment:
If this doesn't exist shouldn't it be a 404?

Reply:
It should be, and we do get a 404, but rops changes it to 500 👎